force-app/main/default/aura/CompensationTable/CompensationTableHelper.js

```diff
@@ -0,0 +1,32 @@
+({
+	getCompensations : function (component) {
+		var action = component.get("c.getCompensations");
+        action.setCallback(this,function(response){
+            if (response.getState() === "SUCCESS"){
+                component.set("v.compensationDetails",response.getReturnValue());
+            }
+            else
+            {
+                alert('Error retrieving compensation details.');
+            }
+        });
+        $A.enqueueAction(action);
+	},
+    filter : function (component){
+        var filter = component.find("select").get("v.value");
+        var rows = document.getElementById("table").rows;
```

Comment:
we can do this better, you can adjust your logic if you sent the value through an event in the UI and just receive the value as a parameter on your method.